Python/Product/VSInterpreters/Interpreter/InterpreterRegistryService.cs

```diff
@@ -86,6 +98,21 @@ sealed class InterpreterRegistryService : IInterpreterRegistryService, IDisposab
             }
         }
 
+        // Called when a single async interpreter factory provider finishes discovering interpreters
+        private void Provider_InterpreterDiscoveryCompleted(object sender, EventArgs e) {
+
+            // Since we know how many async providers we have, keep track of how many times this callback is hit.
+            // Once the number of calls == the number of providers, we know all async interpreter discovery is done.
+            _asyncInterpreterDiscoveryCompletedCount++;
+
+            if (_asyncInterpreterDiscoveryCompletedCount < _asyncProviderCount) {
+                return;
+            }
+
+            // if we get here, interpreter discovery is finished
+            InterpreterDiscoveryCompleted?.Invoke(this, EventArgs.Empty);
```

Comment:
So the `_condaEnvCreateInfoBar.CheckAsync()` is called when the `InterpreterDiscoveryCompleted` event is raised, and there are two scenarios where this event is raised:
1. When `EnsureInitialized()` is called on project load
2. When the `InterpretersChanged` event calls `EnsureFactoryChangesWatched()`, this is where you make sure all the async calls are done before checking the info bar.

Is my understanding correct?